au/wrapper_operations_test.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Aurora Operations, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "au/wrapper_operations.hh"
+
+#include "au/testing.hh"
+#include "au/units/liters.hh"
+#include "au/units/meters.hh"
+#include "au/units/moles.hh"
+#include "gtest/gtest.h"
+
+using ::testing::StaticAssertTypeEq;
+
+namespace au {
+namespace detail {
+namespace {
+
+template <typename Unit>
+struct UnitWrapper : MakesQuantityFromNumber<UnitWrapper, Unit>,
+                     ScalesQuantity<UnitWrapper, Unit>,
+                     ComposesWith<UnitWrapper, Unit, UnitWrapper, UnitWrapper>,
+                     ComposesWith<UnitWrapper, Unit, QuantityMaker, QuantityMaker>,
+                     ForbidsComposingWith<UnitWrapper, Unit, QuantityPointMaker>,
+                     ForbidsComposingWith<UnitWrapper, Unit, QuantityPoint>,
+                     CanScaleByMagnitude<UnitWrapper, Unit> {};
```

Comment:
Note to reviewers: this is probably the best starting point.  The goal of this PR is to enable us to write something like this.  The test cases below demonstrate what capabilities this kind of definition provides.  Then, with that motivation, you can check out the implementations in the previous file.